Packages/Tracking/Physical Hands/Runtime/Scripts/PhysicalHandsButtonToggle.cs

```diff
@@ -0,0 +1,107 @@
+/******************************************************************************
+ * Copyright (C) Ultraleap, Inc. 2011-2024.                                   *
+ *                                                                            *
+ * Use subject to the terms of the Apache License 2.0 available at            *
+ * http://www.apache.org/licenses/LICENSE-2.0, or another agreement           *
+ * between Ultraleap and you, your company or other organization.             *
+ ******************************************************************************/
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Animations;
+using UnityEngine.Events;
+
+namespace Leap.Unity.PhysicalHands
+{
+    /// <summary>
+    /// Class for toggling a button using physical hands.
+    /// </summary>
+    public class PhysicalHandsButtonToggle : PhysicalHandsButtonBase
+    {
+        private bool _pressingObjectExited = false;
+
+        /// <summary>
+        /// Action to perform when the button is pressed.
+        /// </summary>
+        protected override void ButtonPressed()
+        {
+            base.ButtonPressed();
+            // Freeze object's position and rotation when pressed
+            FreezePressableMovement();
+            _pressingObjectExited = false;
+        }
+
+        /// <summary>
+        /// Action to perform when the button is unpressed.
+        /// </summary>
+        protected override void ButtonUnpressed()
+        {
+            base.ButtonUnpressed();
+            // Remove constraints when unpressed
+            UnFreezePressableMovement();
+        }
```

Comment:
I am not getting the expected toggle behavior from these - the buttons don't seem to freeze when i press them down